### PR TITLE
Improvements to the support of reg arrays

### PR DESCRIPTION
**Pull request by eponier**
In particular, it fixes a bad interaction with reg ptr and turns most of the errors into non-internal ones.

Fixes https://github.com/jasmin-lang/jasmin/issues/708

### CODE REVIEW

**Review by bgregoir — APPROVED** [review]
I am not sure that remove the "." in the CHANGELOG is a good idea.
But I do not really care.
Except this point, it is very good.

File: proofs/compiler/array_expansion.v [hunk 1]
```diff
@@ -1,4 +1,4 @@
-(* ** Imports and settings *)
+  (* ** Imports and settings *)
```
Comment (vbgl) [inline proofs/compiler/array_expansion.v:1]: Is this change intended?

**Comment by vbgl** [comment]
I don’t like the last commit.

File: proofs/compiler/array_expansion.v (hunk 1 above)
Comment (eponier) [reply proofs/compiler/array_expansion.v:1]: No.

**Comment by eponier** [comment]
I dropped the last commit to avoid blocking the PR.

**Review by vbgl — APPROVED** [review]
